src/relations/peers.py

```diff
@@ -171,20 +142,9 @@ def _on_changed(self, event: HookEvent):
             self.charm.client_relation.update_endpoints()
 
     def _on_departed(self, event):
-        self.update_leader()
         self.charm.update_client_connection_info()
-        if event.departing_unit == self.leader_unit:
-            self.charm.unit.status = MaintenanceStatus(
-                "Leader unit removed - waiting for leader_elected event"
-            )
```

Comment:
We maintain LEADER_ADDRESS_KEY in the peer data just for this status and I don't think we should be setting it anyway. Removed to bump coverage.